### ISSUE TITLE
Pull in all of UMLS

### ISSUE THREAD

**Issue by cbizon** [TranslatorSRI/Babel#579]
Nodenormalizer has a few roles: 
1. establishing equivalent id sets
2. Assigning types
3. returning labels

One problem is that people rely on it for 3, so if it can't do 1&2 for some chunk of data, then you don't get any labels.

Could we/ should we pull in all of UMLS either as 1) correctly merging everything in there into clique/types or 2) just bringing in the unmapped parts as NamedThings so that we can at least serve labels?

**Comment by gaurav** [comment]
Claude dug into this to figure out which STYs are handled and which aren't, and what the lowest-hanging-fruit fixes are.

## Summary of findings

### T090 (Occupation or Discipline) and T091 (Biomedical Occupation or Discipline) — the real gap

These are the biggest problem. Neither has a Biolink mapping in bmt, and neither has an `STY_OVERRIDE` in `leftover_umls.py`. The current code's conservative behavior: if a concept has **any** unmapped STY, the entire concept is skipped from leftover UMLS. That means:

- Pure T090/T091 concepts → dropped entirely (2,314 + 5,780 ≈ 8,094 KG2 nodes not resolving)
- Any concept with T090 or T091 **plus** other well-mapped STYs (e.g., T090 + T121 Pharmacologic Substance) → also dropped, because T090 is unmapped and blocks the whole concept

**Lowest-hanging-fruit fix:** Add `None` overrides for T090 and T091 in `STY_OVERRIDES`. A `None` override means "deliberately rejected" (not just "unmapped"), which lets the code continue to the concept's other semantic types. Result:
- Pure T090/T091 concepts: still excluded (appropriate — these are research fields/professions, not biomedical entities Babel targets)
- Concepts with T090/T091 + a biomedical STY: now typed by the biomedical STY

This is a ~2-line code change plus test updates. Whether we also want to positively map T090/T091 to `biolink:NamedThing` (to preserve all 8,094 concepts) is a harder policy question I'd put off.

### T021, T045, T120 — already have overrides

These three already have `STY_OVERRIDE` entries in the code (T021 → GrossAnatomicalStructure, T045 → BiologicalProcess, T120 → ChemicalEntity, added in #421). The remaining non-resolving counts (38 T021, 11,132 T045, 871 T120) likely reflect one of:
1. Stale compendium data — the leftover UMLS compendium needs a pipeline rerun to pick up the overrides
2. Concepts already claimed by anatomy/biology/chemistry compendia without their UMLS CURIEs crosslinked (so the UMLS CURIE doesn't appear in any compendium at all)

T045 at 11,132 is surprisingly large; that might be mostly category 2 (GO-mapped biological process concepts where the UMLS CURIE wasn't added to the chemicals/biology clique).

### T121 (Pharmacologic Substance, 188), T123 (Biologically Active Substance, 141), T131 (Hazardous or Poisonous Substance, 49), T061 (Therapeutic or Preventive Procedure, 4)

These already have bmt mappings (T121 → Drug, T123/T131 → ChemicalEntity, T061 → Procedure). They should be in leftover UMLS for concepts not already claimed by other compendia, and leftover UMLS already passes `extra_prefixes=[UMLS]` to `write_compendium()`, so UMLS CURIEs are accepted even for types that don't list UMLS in Biolink's `id_prefixes`.

The residual non-resolving counts most likely represent UMLS CURIEs that appear in the chemicals/procedure compendia (via ChEBI, DrugBank, RXNORM, CPT crosslinks) but where the UMLS CURIE itself wasn't added to the clique's identifier list. That's a deeper cross-referencing gap, not a leftover UMLS issue. Probably not worth fixing now.

### T043 (Cell Function, 1), T042 (Organ or Tissue Function, 2), T170 (Intellectual Product, 3), T201 (Clinical Attribute, 1), T098 (Population Group, 1), T069 (Environmental Effect of Humans, 1), T078/T169 (Idea or Concept/Functional Concept, 1 each)

All have bmt mappings. Single-digit non-resolving counts. Probably edge cases or specific CURIEs that ended up in other compendia without UMLS crosslinks. Not worth investigating now.

## Proposed next steps

1. **Do now (small):** Add `None` overrides for T090 and T091 in `STY_OVERRIDES` in `src/createcompendia/leftover_umls.py`, with a test update. This unblocks any multi-typed concepts and makes the behavior explicit. (See https://github.com/NCATSTranslator/Babel/issues/817 for the more comprehensive solution)
2. **After pipeline rerun:** Check whether the T021/T045/T120 non-resolving counts drop significantly — that would confirm the stale-data hypothesis.
3. **Later (bigger):** If T045 (Genetic Function, 11k) is still large after a rerun, investigate which specific UMLS CURIEs aren't in any compendium and whether they should be added to the biology/process compendia's cross-referencing.
4. **Policy question (defer):** Should T090/T091 concepts map to `biolink:NamedThing` so they at least get labels? Needs consensus on whether occupations/disciplines belong in the graph.

**Comment by gaurav** [comment]
AFAICT, with the Babel 1.17 branch (PR https://github.com/NCATSTranslator/Babel/pull/824), every currently valid UMLS identifier is now in Babel. Closing. Please open a new issue if you find UMLS identifiers that are missing from Babel.
